### PR TITLE
PMM-5968 Avoid removing PMM-Server.

### PR DESCRIPTION
**Pull request by BupycHuk**
https://github.com/Percona-Lab/pmm-submodules/pull/874

### CODE REVIEW

File: models/agent_helpers.go [hunk 1]
```diff
@@ -447,6 +447,10 @@ func RemoveAgent(q *reform.Querier, id string, mode RemoveMode) (*Agent, error)
 		return nil, err
 	}
 
+	if id == PMMServerAgentID {
+		return nil, status.Error(codes.PermissionDenied, "Removing pmm-agent on PMM Server prevents remote monitoring.")
```
Comment (askomorokhov) [inline models/agent_helpers.go:451]: But does not this agent monitor **local** PostgreSQL by default as well?
If yes, maybe we need a more general error message here.

File: models/node_helpers.go [hunk 2]
```diff
@@ -247,6 +247,10 @@ func RemoveNode(q *reform.Querier, id string, mode RemoveMode) error {
 		return err
 	}
 
+	if id == PMMServerNodeID {
+		return status.Error(codes.PermissionDenied, "Removing pmm-server node prevents remote monitoring.")
```
Comment (askomorokhov) [inline models/node_helpers.go:251]: same question as above.

File: models/agent_helpers.go (hunk 1 above)
Comment (BupycHuk) [reply models/agent_helpers.go:451]: Yes, it is, but it's not critical.

**Comment by BupycHuk** [comment]
@AlekSi ok, but what is wrong with this PR? Could you point what should be fixed?

**Comment by AlekSi** [comment]
I could, but I would prefer to address the root cause and discuss solutions, not "just fix" that PR.
The fix should not take too much time and does not block QA, etc.

**Comment by BupycHuk** [comment]
@AlekSi https://github.com/Percona-Lab/pmm-submodules/pull/874

**Comment by AlekSi** [comment]
/s/have/update. API tests fail there because it wasn't updated properly

**Pull request merged**